lib/_http_client.js

```diff
@@ -97,27 +96,12 @@ class HTTPClientAsyncResource {
   }
 }
 
-let urlWarningEmitted = false;
 function ClientRequest(input, options, cb) {
   FunctionPrototypeCall(OutgoingMessage, this);
 
   if (typeof input === 'string') {
     const urlStr = input;
-    try {
-      input = urlToOptions(new URL(urlStr));
-    } catch (err) {
-      input = url.parse(urlStr);
-      if (!input.hostname) {
-        throw err;
-      }
-      if (!urlWarningEmitted && !process.noDeprecation) {
-        urlWarningEmitted = true;
-        process.emitWarning(
-          `The provided URL ${urlStr} is not a valid URL, and is supported ` +
-          'in the http module solely for compatibility.',
-          'DeprecationWarning', 'DEP0109');
```

Comment:
If we're removing this, shouldn't DEP0109 be updated in doc/api/deprecations.md to be EOL?

Reply:
Thanks for mentioning. I changed it. :)